### ISSUE TITLE
Make `nvrtc.getCUBIN()` default on CUDA 11.1+?

### ISSUE THREAD

**Issue by leofang** [cupy/cupy#4591]
In #4558 the new `nvrtcGetCUBIN()` API was exposed. The purpose was to avoid the driver from performing just-in-time compilation of PTX codes, which [was shown slow unless the jitted kernels are cached](https://github.com/cupy/cupy/pull/4558#issuecomment-765610834) (CUDA's default behavior). However, we decided to temporarily disable this feature in 8cb443f. This issue is for discussing the strategy to utilize this feature, which IMHO is a promising one 🙂

**Comment by leofang** [comment]
FWIW, when determining the performance of `getCUBIN()` it's best to set `CUDA_CACHE_DISABLE=1`. For fair comparison, it's important to always force JIT compiling PTX instead of fetching from cache. It is expected to benefit fresh kernels.

**Comment by leofang** [comment]
If `getCUBIN()` is enabled by default, this problem would be avoided: https://stackoverflow.com/a/66788739/2344149

tl;dr: To keep the kernel cache valid (without recompiling), users not only have to keep `~/.cupy/kernel_cache` but also `~/.nv/ComputeCache`, because CuPy only gets and caches the PTXs from `getPTX()`.
